frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -315,3 +332,20 @@ function getBaseOrigin(location: Location): string {
    }
    return location.origin;
 }
+
+function getDevicePath(handle: string, routeState: RouteState): string {
+   const [device, variant] = handle.split(':');
+   const variantFromRouteState = routeState.filter?.['worksin'];
+
+   if (variant && !variantFromRouteState) {
+      return device;
+   }
+
+   if (!variant && variantFromRouteState) {
+      const title = destylizeDeviceTitle(handle);
+      const varaintToAdd = variantFromRouteState.replace(title, '').trim();
+      return `${handle}:${varaintToAdd}`;
```

Comment:
Very small typo, not worth changing unless you're making other changes
```suggestion
      const variantToAdd = variantFromRouteState.replace(title, '').trim();
      return `${handle}:${variantToAdd}`;
```